### ISSUE TITLE
Investigate options for registry cleanup

### ISSUE THREAD

**Issue by arboleya** [FuelLabs/fuels-ts#3017]
> _This is a follow-up issue after https://github.com/FuelLabs/fuels-ts/issues/2377_

When someone runs `npm info fuels,` the output is far from ideal:

<details>
  <summary>Click for screenshot</summary>

![npm-info-fuels](https://github.com/user-attachments/assets/a84ac437-3d9b-42c5-b866-b2a78a85b0d1)

</details>

The solution to this would include two steps:
 - [x] Contact NPM support to remove all irrelevant tags
 - [X] Deprecate irrelevant tags
 - [x] Stop publishing temporary tags to NPM

The ability to publish temporary PRs is helpful, but it'd be great if we could do that separately from the NPM registry.

Could we use GitHub packages to resolve the second point?
 - https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages

Ideally, we'd still be able to publish temp tags while keeping our packages clean and tidy.

**Comment by petertonysmith94** [comment]
@danielbate I think the second point here is super valuable, as it'd enable us to run the UI integration tests by default, and not only on merge to master.

**Comment by arboleya** [comment]
Also, here's the NPM policy:
https://docs.npmjs.com/policies/unpublish

**Comment by danielbate** [comment]
I have contacted support to remove the [following versions](https://gist.github.com/danielbate/ed8839088c2f48b456f5ff1ba9446fdf), let's see how it goes. Worst case scenario we can alter the script and [deprecate](https://docs.npmjs.com/cli/v8/commands/npm-deprecate) these versions.

**Comment by arboleya** [comment]
Easy, _just_ 21k+ versions. 😅

**Comment by danielbate** [comment]
Okay so unfortunately support cannot remove our versions... Our weekly download count is too high and we have too many dependents. 

I'm going to make an adjustment to the script so that it deprecates these versions rather than attempting to un-publish. I'd also favour having it run on a cron whilst we go through this process.

I'll also look to move future published PR versions to a seperate registry.

**Comment by danielbate** [comment]
Deprecation will clean up `info` alot, this is the result for nextjs who have more published versions than us, but deprecate.

<img width="809" alt="Screenshot 2024-12-11 at 11 03 02" src="https://github.com/user-attachments/assets/b0c82fb8-6df0-451b-b9b5-c8daf93cab88">

**Comment by danielbate** [comment]
@arboleya I've investigated using a private registry and it certainly is possible. However one reason we use PR versions is for testing with consumers. Using a private/separate registry will require the consumer to configure the `.npmrc` file to load from the registry.

I think this problem could be mitigated if we just correctly and regularly deprecate versions. Once #3465 is through I'll make a change so we can pass singular versions, and maybe even a cron that cleans up versions every few days (but I'm not sold on the latter as consumers take time to respond). Do you think this would suffice, or still prefer a separate registry?

**Comment by arboleya** [comment]
@danielbate Could we deprecate a specific tag when the related PR goes to `master`?

Can custom registries be specified as a CLI option or just as a `.npmrc` config?

```bash
npm install abc --registry=xyz
```

Either way, it's perfectly fine to move in small increments here.

We can start with deprecation and still consider possible future improvements.

**Comment by danielbate** [comment]
> Could we deprecate a specific tag when the related PR goes to master?

Yes exactly my thoughts. So on a feature PR merge, we could deprecate any associated `pr` release tags. On a release PR merge, we deprecate any previous `next` release tags. I definitely think there's room for automated deprecations to manage this.

AFAIK it's only via the `npmrc` for a custom registry on install, but I will see.

**Comment by arboleya** [comment]
@danielbate We will ***stop*** publishing non-tag versions to NPM, effective immediately. For temporary and non-tagged releases *(*`pr-*`*,* `next`*)*, we must use another registry like GitHub. Bumping priority on this.

**Comment by arboleya** [comment]
@danielbate We will ***stop*** publishing non-tag versions to NPM, effective immediately. For temporary and non-tagged releases *(*`pr-*`*,* `next`*)*, we must use another registry like GitHub. Bumping priority on this.

**Comment by danielbate** [comment]
Makes sense. Given the increase in priority I've unassigned my self for now. Hopefully it can be picked up sooner by either @petertonysmith94 or @nedsalk?

**Comment by arboleya** [comment]
Great job, man! Record time. 👏 

I don't see anything wrong with Github - do you?

I think the flow looks something like this:

```mermaid
---
title: Release Flow
---
flowchart
  PR([PR #4876]) --> Master(master)
  PR --> |‎ ‎ pr-4876‎‎ ‎ | GH
  Master --> IsRelease[[is this a release?]]
  IsRelease --> Yes(yes)
  IsRelease --> No(no)
  Yes --> |‎ ‎ latest‎‎ ‎ | Npm{NPM}
  No --> |‎ ‎ next‎‎ ‎ | GH(((GitHub)))
```

**Comment by petertonysmith94** [comment]
We have managed to deprecate all our `0.0.0-x` versions on the following action:
- https://github.com/FuelLabs/fuels-ts/actions/runs/13184120371/job/36802008166#step:6:42

Using the following PR:
- https://github.com/FuelLabs/fuels-ts/pull/3670

---

We've have a PR to integration GitHub packages:
- https://github.com/FuelLabs/fuels-ts/pull/3639

I've created an issue to enhance this workflow further:
- https://github.com/FuelLabs/fuels-ts/issues/3671

**Comment by petertonysmith94** [comment]
Closed via:

* #3670

**Comment by petertonysmith94** [comment]
Closed via:

* #3670
